suave/sdk/sdk.go

```diff
@@ -64,29 +68,76 @@ func (c *Contract) SendTransaction(method string, args []interface{}, confidenti
 		return nil, err
 	}
 
-	computeRequest, err := types.SignTx(types.NewTx(&types.ConfidentialComputeRequest{
-		ConfidentialComputeRecord: types.ConfidentialComputeRecord{
-			KettleAddress: c.client.kettleAddress,
-			Nonce:         nonce,
-			To:            &c.addr,
-			Value:         nil,
-			GasPrice:      gasPrice,
-			Gas:           1000000,
-			Data:          calldata,
-		},
-		ConfidentialInputs: confidentialDataBytes,
-	}), signer, c.client.key)
+	record := &types.ConfidentialComputeRecord{
+		KettleAddress:          c.client.kettleAddress,
+		Nonce:                  nonce,
+		To:                     &c.addr,
+		Value:                  big.NewInt(0),
+		GasPrice:               gasPrice,
+		Gas:                    1000000,
+		Data:                   calldata,
+		ConfidentialInputsHash: crypto.Keccak256Hash(confidentialDataBytes), // ?
+
+		// initialize empty fields
+		ChainID: big.NewInt(1337), // suave chain id?
+		V:       big.NewInt(0),
+		R:       big.NewInt(0),
+		S:       big.NewInt(0),
+	}
+
+	/*
+		computeRequest, err := types.SignTx(types.NewTx(&types.ConfidentialComputeRequest{
+			ConfidentialComputeRecord: *record,
+			ConfidentialInputs:        confidentialDataBytes,
+		}), signer, c.client.key)
+		if err != nil {
+			return nil, err
+		}
+	*/
+
+	domain := &eip712.EIP712Domain{
+		Name:              "suave?",
+		Version:           "1.0",
+		VerifyingContract: "0x0000000000000000000000000000000000000000", // ??
+		ChainId:           big.NewInt(1),                                // suave chain id?
+		Salt:              []byte{0x1, 0x2},
+	}
+	builder := eip712.NewEIP712MessageBuilder[types.ConfidentialComputeRecord](domain)
+	typedData := builder.Build(record)
+
+	typedDataHashed, err := typedData.Hash()
 	if err != nil {
-		return nil, err
+		panic(err)
 	}
 
-	computeRequestBytes, err := computeRequest.MarshalBinary()
+	signedMsg, err := c.client.Sign(typedDataHashed[:])
```

Comment:
So this does not work. `suaveTransaction` uses the signature values from record (`v`, `r`, `s` and `chainid`) to validate the transaction. But, those fields are not populated.
But, can they be populated? Not really, because you are signing the EIP-712 message and not the record. So, I think we have to move the signature values to the outer envelope of the EIP-712, thoughts?